content/blogs/2024-04-05-getting-started-with-kestra.md

```diff
@@ -0,0 +1,315 @@
+  image: "wrussell"
+image: /blogs/2024-04-05-getting-started-with-kestra.png
+---
+
+Kestra is an event driven data orchestration platform that’s highly flexible and easy to use. This guide is going to go through the basics and get you started building your own pipeline!
+
+Originating as a platform for data orchestration, Kestra finds itself well-equipped to manage all types of pipelines with its highly flexible interface and a huge range of plugins. Through this blog post, I’m going to show you how to get setup with Kestra and set up a simple workflow to run a Python script every hour and send the result as a Discord notification.
+
+## Installation
+
+Kestra is open source meaning anyone can run it on their machine for free. There is a cloud version coming soon for those who want to start making flows straight away, however for now we'll set it up locally. To get it setup, you’ll need to make sure you have Docker installed on your machine and run the following command to start up your instance!
+
+```bash
+docker run --pull=always --rm -it -p 8080:8080 --user=root \
+  -v /var/run/docker.sock:/var/run/docker.sock \
+  -v /tmp:/tmp kestra/kestra:latest-full server local 
+```
+
+Once you’ve run this command, head over to your browser and open [https://localhost:8080](https://localhost:8080) to launch the interface so we can start building workflows!
+
+## Properties
+
+Before we start making our workflows, it’s worth understand a few fundamental properties that we’ll use to build everything!
+
+Workflows are referenced as Flows and they’re declared using YAML making it very readable as well as works with any language! Within each flow, there are 3 required properties you’ll need:
+
+- `id` which is the name of your flow. This can’t be changed once you’ve executed your flow for the first time.
+- `namespace` which allows you to specify the environments you want your flow to execute in, e.g. production vs development
+- `tasks` which is a list of the tasks that will execute when the flow is executed, in the order they’re defined in. Tasks contain an `id` as well as a `type` with each different type having their own additional properties.
+
+To help visualise it, here's an example:
+
+```yaml
+id: getting_started
+namespace: example
+tasks:
+  - id: hello_world
+    type: io.kestra.core.tasks.log.Log
+    message: Hello World!
+```
+
+Everything builds off of these 3 properties but there’s a few more optional properties that you’ll want to use to give you full flexibility.
+
+- **Inputs:** Instead of hardcoding values into your flows, you can set them as constant values separately. Great if you plan to reuse them in multiple tasks. An input might look like this: `{{ inputs.webhook_url }}` .
+- **Outputs:** Tasks will often generate outputs that you’ll want to pass on to a later task. Outputs let you connect both variables as well as files to later tasks. An output of a variable could look like this: `"{{ outputs.script.vars['output'] }}"`
```

Comment:
```suggestion
- **Outputs:** Tasks will often generate outputs that you’ll want to pass on to a later task. Outputs let you connect both variables as well as files to later tasks. An output of a variable could look like this: `"{{ outputs.script.vars.myoutput }}"`
```

a matter of preference but I'd only use the subscript notation [] only if variable name has hyphens

Reply:
Ah didn't realise you could also do it this way. I'll update to reflect that as it's much clearer. Also making notes for the docs improvements as I struggled to understand the syntax for outputs for Python.